src/frontend/js/modules/authentication.js

```diff
@@ -7,16 +7,39 @@ angular.module('esn.authentication', ['esn.http'])
     const SERVER_TTL = 60 * 1000;
     const NETWORK_LAG = 15 * 1000;
     const CACHE_MAX_TTL = SERVER_TTL - NETWORK_LAG;
+    const JWT_CACHE_MAX_TTL = 60 * 1000 * 15; // 15 minutes
     const cachedToken = { promise: null, timestamp: 0 };
+    const cachedJWTToken = { promise: null, timestamp: 0 };
 
     return {
-      getNewToken
+      getNewToken,
+      getWebToken
     };
 
     function getNewToken(resetCache = false) {
       return resetCache ? getTokenNoCache() : getCachedToken();
     }
 
+    function getWebToken(reset = false) {
+      return reset ? requestNewWebToken() : getCachedWebToken();
+    }
+
+    function requestNewWebToken() {
+      setJWTCache(esnRestangular.one('jwt/generate').post(), Date.now());
+
+      cachedJWTToken.promise.catch(() => {
+        setJWTCache(null);
+      });
+
+      return cachedJWTToken.promise;
+    }
+
+    function getCachedWebToken() {
```

Comment:
Hello, There is a `Cache` service in the ESN frontend which you can use and which provide exactly what you need

Reply:
thanks